tuf/src/metadata.rs

```diff
@@ -1457,69 +1457,40 @@ impl VirtualTargetPath {
         self.matches_chain(&*new)
     }
 
+    /// Prefix the target path with a hash value to support TUF spec 5.5.2.
+    pub fn with_hash_prefix(&self, hash: &HashValue) -> Result<TargetPath> {
+        let mut components = self.components();
+
+        // The unwrap here is safe because we checked in `safe_path` that the path is not empty.
```

Comment:
Though why not just return an error if the pop fails? A little belt-and suspenders doesn't hurt here.

Reply:
good idea, added.